src/SFA.DAS.AssessorService.Application/Handlers/Certificates/UpdateCertificatesPrintStatusHandler.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.AssessorService.Api.Types.Models;
+using SFA.DAS.AssessorService.Api.Types.Models.Certificates;
+using SFA.DAS.AssessorService.Api.Types.Models.Validation;
+using SFA.DAS.AssessorService.Application.Interfaces;
+using SFA.DAS.AssessorService.Domain.Consts;
+using SFA.DAS.AssessorService.Domain.Extensions;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.Certificates
+{
+    public class UpdateCertificatesPrintStatusHandler : IRequestHandler<UpdateCertificatesPrintStatusRequest, ValidationResponse>
+    {
+        private readonly ICertificateRepository _certificateRepository;
+        private readonly IMediator _mediator;
+        private readonly ILogger<UpdateCertificatesPrintStatusHandler> _logger;
+
+        public UpdateCertificatesPrintStatusHandler(ICertificateRepository certificateRepository, IMediator mediator, ILogger<UpdateCertificatesPrintStatusHandler> logger)
+        {
+            _certificateRepository = certificateRepository;
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        public async Task<ValidationResponse> Handle(UpdateCertificatesPrintStatusRequest request, CancellationToken cancellationToken)
+        {
+            var validationResult = new ValidationResponse();
+
+            var notFoundBatches = new List<int>();
+            var validPrintStatus = new List<string> { CertificateStatus.Printed, CertificateStatus.Delivered, CertificateStatus.NotDelivered };
+            
+            foreach (var certificatePrintStatus in request.CertificatePrintStatuses)
+            {
+                if (!notFoundBatches.Contains(certificatePrintStatus.BatchNumber))
+                {
+                    if(await _mediator.Send(new GetForBatchNumberBatchLogRequest { BatchNumber = certificatePrintStatus.BatchNumber }) == null)
+                    {
+                        validationResult.Errors.Add(new ValidationErrorDetail(nameof(request.CertificatePrintStatuses), $"The batch number {certificatePrintStatus.BatchNumber} was not found.", ValidationStatusCode.NotFound));
+                        notFoundBatches.Add(certificatePrintStatus.BatchNumber);
+                    }
+                }
+
+                if (!notFoundBatches.Contains(certificatePrintStatus.BatchNumber))
```

Comment:
same if statement as above, this code could be a continuation of the above block with a `continue;` after the `notFoundBatches.Add()`